src/hooks/custom/useInsightInput.ts

```diff
@@ -0,0 +1,54 @@
+import { InsightPostRequest } from '@/types/insight';
+import { useCheckMainImage } from '@/utils/upload';
+import { useState } from 'react';
+import defaultImage from '@image/defaultImage.jpeg';
+import { ParsedUrlQuery } from 'querystring';
+
+interface InsightSummary {
+  title: string | undefined;
+  summary: string | undefined;
+  keywords: string | undefined;
+  folderName: string[] | undefined;
+}
+
+export default function useInsightInput(query: ParsedUrlQuery) {
+  const image = useCheckMainImage(query?.imageList, String(query?.link));
+
+  const [insightInput, setInsightInput] = useState<InsightPostRequest>({
+    insightUrl: '',
+    insightTitle: '',
+    insightSummary: '',
+    insightMainImage: image ?? defaultImage.src,
+    insightSource: String(query?.source),
+    viewCount: 0,
+    insightTagList: [''],
+    insightMemo: String(query?.memo),
```

Comment:
위에서 destructure 하고 alias 까지 지정하면 더 간단해 질 것 같습니다.

```suggestion
    insightMemo,
```